dbms/src/Debug/MockExecutor/WindowBinder.cpp

```diff
@@ -121,25 +121,43 @@ bool WindowBinder::toTiPBExecutor(tipb::Executor * tipb_executor, int32_t collat
         if (frame.start.has_value())
         {
             auto * start = mut_frame->mutable_start();
-            start->set_offset(std::get<2>(frame.start.value()));
-            start->set_unbounded(std::get<1>(frame.start.value()));
-            start->set_type(std::get<0>(frame.start.value()));
+            start->set_offset(frame.start->getOffset());
+            start->set_unbounded(frame.start->isUnbounded());
+            start->set_type(frame.start->getBoundType());
+            if (frame.start->isRangeFrame())
+            {
+                // It's ok to pass raw pointer here because the destruction of
+                // mock::MockWindowFrame happens after the execution of DAGRequest
+                // and we can ensure that the raw pointer is always valid in the
+                // process of executing.
+                start->set_allocated_frame_range(frame.start->robRangeFrame());
+                start->set_cmp_data_type(frame.start->getCmpDataType());
```

Comment:
a little hack, can you try a deep copy?

Reply:
> a little hack, can you try a deep copy?

Comment is useless and I forget to delete it. Here we rob the nullptr, and the pointer is held by only one object, it's not hack.

Reply:
> a little hack, can you try a deep copy?

The comment is useless and I forget to delete it. After `robRangeFrame()`, the pointer will be held only by one object, it's safe and not hack.